core/uploadDiveShare.cpp

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "uploadDiveShare.h"
+#include <QDebug>
+#include "core/membuffer.h"
+#include "core/settings/qPrefCloudStorage.h"
+#include "core/qthelper.h"
+#include "core/cloudstorage.h"
+#include "core/save-html.h"
+#include "core/errorhelper.h"
+
+
+uploadDiveShare *uploadDiveShare::instance()
+{
+	static uploadDiveShare *self = new uploadDiveShare;
+
+	return self;
+}
+
+
+uploadDiveShare::uploadDiveShare():
+	reply(NULL)
+{
+	timeout.setSingleShot(true);
+}
+
+
+void uploadDiveShare::doUpload(bool selected, const QString &uid, bool noPublic)
+{
+}
+
+
+void uploadDiveShare::slot_updateProgress(qint64 current, qint64 total)
```

Comment:
silly bike-shedding comment about the name. Again, no need to refactor for this, can always simply be renamed later.